### PR TITLE
[FEATURE] Detect the type of resource field for file fields

### PR DESCRIPTION
**Pull request by e0ipso**
Set the class to ResourceFieldFileEntityReference if there is an
associated resource in the field definition. That allows embedding
resources easily.

### CODE REVIEW

File: src/Plugin/resource/Field/ResourceFieldEntity.php [hunk 1]
```diff
@@ -890,6 +890,10 @@ public static function fieldClassName(array $field_definition) {
       case 'text':
       case 'text_long':
       case 'text_with_summary':
+        // If the field is treated as a resource, then default to the reference.
+        if (!empty($field_definition['resource'])) {
```
Comment (amitaibu) [inline src/Plugin/resource/Field/ResourceFieldEntity.php]: resource that is a of type text\* - how come?

Comment (e0ipso) [inline src/Plugin/resource/Field/ResourceFieldEntity.php]: :weary: I had to strip this out from an ongoing PR. This should go to the File field.

Good :eyes:

Comment (amitaibu) [inline src/Plugin/resource/Field/ResourceFieldEntity.php]: I IS SMARTS! ;)

**Comment by e0ipso** [comment]
Please @amitaibu use merge --squash. I'm trying to keep the commit history clean with release notes valuable commit messages.

I'm providing the desired commit message as the first commit in the PR (also used as title and description of the PR). The format is inspired by https://wiki.typo3.org/CommitMessage_Format_(Git).

**Comment by amitaibu** [comment]
> I'm trying to keep the commit history clean with release notes valuable commit messages.

You can also `git rebase -i origin/7.x-2.x` and mark all the commits, apart of the first one as `f`

**Comment by amitaibu** [comment]
Merged, thanks.

**Pull request closed without merging**